auth/design.md

```diff
@@ -0,0 +1,69 @@
+﻿
+# Design document of the authentication mechanism in etcd v3 protocol
+
+## Why not using the existing v2 auth mechanism?
+The v3 protocol uses gRPC as its transport instead of RESTful interface like v2. Therefore we have a chance of improvements (e.g. connection based auth instead of request based auth). In addition, there is an unclear point in the semantics of the v2 auth (described later). In v3, it should be cleaned and a well defined guidance should be provided for users and developers.

```

Comment:
The v3 protocol uses gRPC as its transport instead of RESTful interface like v2. This new protocol provides an opportunity to iterate on and improve the v2 design. For example, v3 auth has connection based authentication, rather than v2's slower per-request authentication. Additionally, v2 auth's semantics tend to be unwieldy in practice with respect to reasoning about consistency, which will be described in the next sections. For v3, there is a well-defined description and implementation of the authentication mechanism which fixes the deficiencies in the v2 auth system.